multiverse/tipmanager.go

```diff
@@ -221,9 +224,36 @@ func (URTS) TipSelect(tips *randommap.RandomMap, maxAmount int) []interface{} {
 }
 
 // TipSelect selects maxAmount tips
-// TODO: Modify this tip selection algorithm
 // RURTS: URTS with max parent age restriction
 func (RURTS) TipSelect(tips *randommap.RandomMap, maxAmount int) []interface{} {
-	return tips.RandomUniqueEntries(maxAmount)
+
+	var tipsToReturn []interface{}
+	// Prune the old tips
+	for {
+		tipsToReturn = tips.RandomUniqueEntries(maxAmount)
+
+		// If no tips, return empty list directly
+		if len(tipsToReturn) == 0 {
+			break
+		}
+
+		// Get the current time
+		currentTime := time.Now()
+		pruned := false
+		for _, tip := range tipsToReturn {
+			// If the time difference is greater than DeltaURTS, delete it from tips
+			if currentTime.Sub(tip.(*Message).IssuanceTime).Seconds() > config.DeltaURTS {
+				tips.Delete(tip)
+				pruned = true
+			}
+		}
+
+		// If some tips are pruned, we select again, else we return the selected tips
+		if pruned == false {
```

Comment:
This works but it might elongate the time of the tipselection, becuase you are throwing away the good tips you select... Making you start from scratch, increasing the chances you will pick bad tips again and loop again..

maybe a better strategy would be to get a random entry, if it is good save it, otherwise dump it from `tipsToReturn` and `tips`

I am not 100% confident in what I am saying, if you believe it won't affect the runtime too much feel free to ignore my comment

Reply:
It makes sense to prevent reselecting. I modified the implementation in this [commit](https://github.com/iotaledger/multiverse-simulation/pull/5/commits/899f8dc03fc7cb4de7caf46a44836563f92caa11). In the commit, the valid tips will be retained, and only when aged tips are selected we will sample again.